src/main/java/com/musala/sdcs/jdbc/DeviceRepository.java

```diff
@@ -0,0 +1,210 @@
+import com.musala.sdcs.device.channel.base.ChannelCreator;
+
+/**
+ * This class takes care of DB queries related to devices
+ *
+ */
+public class DeviceRepository {
+	private static final String QUERY_GET_DEVICES = "SELECT * FROM devices;";
+	private static final String QUERY_GET_DEVICES_CHANNELS = "SELECT * FROM devices_channels WHERE deviceId = %s;";
+	private static final String QUERY_GET_CHANNELS = "SELECT * FROM channels WHERE id = %s;";
+	private static final String QUERY_GET_CHANNEL_TYPE = "SELECT * FROM channel_types WHERE id = %s";
+	private static final String QUERY_GET_MODEL = "SELECT * FROM models WHERE id = %s;";
+	private static final String QUERY_GET_MANUFACTURER = "SELECT * FROM manufacturers WHERE id = %s;";
+	
+	private Connection dbConnection;
+
+	public DeviceRepository() {
+		this.dbConnection = DbUtil.getConnection();
+	}
+
+	public List<Device> getAllDevices() {
+		List<Device> devices = new ArrayList<Device>();
+		ResultSet r = executeQuery(QUERY_GET_DEVICES);
+		
+		try {
+			while (r.next()) {
+				Integer deviceId = r.getInt("id");
+				String serialNumber = r.getString("serialNumber");
+				String firmwareVersion = r.getString("firmwareVersion");
+				Integer hardwareVersion = r.getInt("hardwareVersion");
+				String label = r.getString("label");
+				Integer modelId = r.getInt("modelId");
+	
+				List<Channel> channels = getDeviceChannels(deviceId);
+				Map<String, String> modelAndManufacturer = getModelAndManufacturer(modelId);
+	
+				String model = modelAndManufacturer.get("model");
+				String manufacturer = modelAndManufacturer.get("manufacturer");
+	
+				Device device = DeviceBuilderImpl.getInstance().withLabel(label).withId(deviceId).withFirmwareVersion(firmwareVersion)
+						.withHardwareVersion(hardwareVersion).withManufacturer(manufacturer).withModelId(model).withSerialNumber(serialNumber)
+						.withChannels(channels).build();
+				
+				devices.add(device);
+			}
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+		return devices;
+	}
+
+	/**
+	 * @param deviceId
+	 * @return List of channels
+	 */
+	private List<Channel> getDeviceChannels(Integer deviceId) {
+		List<Channel> channels = new ArrayList<Channel>();
+		String query = String.format(QUERY_GET_DEVICES_CHANNELS, deviceId);
+
+		
+			ResultSet r = executeQuery(query);
+
+			try {
+				while (r.next()) {
+					Integer channelId = r.getInt("channelId");
+
+					channels.add(getChannel(channelId));
+				}
+			} catch (SQLException e) {
+				e.printStackTrace();
+			}
+
+		return channels;
+	}
+
+	/**
+	 * @param channelId
+	 * @return Channel
+	 */
+	private Channel getChannel(Integer channelId) {
+		String query = String.format(QUERY_GET_CHANNELS, channelId);
+		
+		
+			ResultSet r = executeQuery(query);
+
+			try {
+				if (r.next()) {
+					Integer id = r.getInt("id");
+					Integer channelTypeId = r.getInt("channelTypeId");
+					String label = r.getString("label");
+					String command = r.getString("command");
```

Comment:
For the whole class - all DB column names must be extracted into constants, best would be to put them in the `DBUtil` class.